overlord/snapstate/handlers_link_test.go

```diff
@@ -1397,6 +1423,213 @@ func (s *linkSnapSuite) TestDoLinkSnapdRemovesAppArmorProfilesOnSnapdDowngrade(c
 	c.Check(lp.instanceNames, DeepEquals, []string{"snapd"})
 }
 
+func (s *linkSnapSuite) TestDoLinkSnapMaybeUninhibitSnapOn(c *C) {
+	s.state.Lock()
+	t := s.state.NewTask("link-snap", "test")
+	sinfo := snap.SideInfo{
+		RealName: "foo",
+		Revision: snap.R(33),
+		SnapID:   "foo-id",
+	}
+	t.Set("snap-setup", &snapstate.SnapSetup{
+		SideInfo: &sinfo,
+		Channel:  "beta",
+		UserID:   2,
+	})
+	s.state.NewChange("sample", "...").AddTask(t)
+
+	s.state.Unlock()
+
+	s.se.Ensure()
+	s.se.Wait()
+
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	// link-snap task should detect that we don't have upcoming
+	// uninhibit-snap tasks in the current change and call
+	// backend.UninhibitSnap itself
+	expected := fakeOps{
+		{
+			op:    "candidate",
+			sinfo: sinfo,
+		},
+		{
+			op:   "link-snap",
+			path: filepath.Join(dirs.SnapMountDir, "foo/33"),
+		},
+		{
+			op:   "uninhibit-snap",
+			name: "foo",
+		},
+	}
+	c.Check(s.fakeBackend.ops.Ops(), DeepEquals, expected.Ops())
+	c.Check(s.fakeBackend.ops, DeepEquals, expected)
+}
+
+func (s *linkSnapSuite) TestDoLinkSnapMaybeUninhibitSnapOffNoraml(c *C) {
```

Comment:
```suggestion
func (s *linkSnapSuite) TestDoLinkSnapMaybeUninhibitSnapOffNormal(c *C) {
```

Reply:
Do you suggest this is renamed to something else?

Reply:
My suggestion was just to fix the typo in "Normal", I think it's fine otherwise

Reply:
My eyes failed me here :D, thanks!